src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -14,13 +18,62 @@
   "v3")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn class-info-from-jruby->class-info-for-json
+  "Convert a class info map received from the jruby service into an
+  appropriate map for use in serializing an environment_classes response to
+  JSON"
+  [info-from-jruby environment]
```

Comment:
I should be able to do that.  I assume you're thinking that we'd have both a schema for the incoming format and a schema for the translated format - to be written to JSON.  I'd resisted doing that initially because of minor concerns about the performance overhead that might come from having it be on during production given that we could be dealing with maps that have hundreds? of objects in them.

Beyond the potential performance concern, the one part I was a little concerned about schematizing is the `default_literal`.   The format of these objects could be fairly variable - and could change in the future as the underlying Puppet language evolves.  I had been hoping that Puppet Server's knowledge of that could be fairly opaque, leaving it to the `ClassInformationService` in Puppet Ruby code to produce either a more primitive object - Integer, String - or tree of primitive objects that will serialize cleanly to JSON.  If we want a more comprehensive schema at this level, though, arguably we should also add some more tests in this project to exercise the different formats that could be returned.

WDYT?